### PR TITLE
fix(inventory): create a management port if the IP is unknown from the list of network port IPs

### PR DESCRIPTION
**Pull request by stonebuzz**
Problem occurs only from Printer (for now)

Even if IP from XML tag ```IPS```  is known from list of network port IPs
GLPI add it as ```port management```

This results in a duplication of the IP and MAC from Printer list (column ```Network - IP``` and ```Nertwork - MAC ```)

To prevent duplicate data, we check if  IP from XML tag ```IPS``` is known from the list of network port IPs
If yes, this ```IP``` is remove from port management list.

see: 
https://github.com/glpi-project/glpi-inventory-plugin/issues/106
https://github.com/glpi-project/glpi-inventory-plugin/issues/183
https://github.com/glpi-project/glpi-inventory-plugin/issues/188

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: src/Inventory/Asset/Printer.php [hunk 1]
```diff
@@ -153,6 +153,31 @@ public function prepare(): array
             }
         }
 
+        //try to know if management port IP is already known as IP port
+        //if yes remove it from management port
+        $port_managment = $this->getManagementPorts();
+        if (isset($this->getManagementPorts()['management']) && property_exists($this->getManagementPorts()['management'], 'ipaddress')) {
```
Comment (trasher) [inline src/Inventory/Asset/Printer.php:159]:
```suggestion
        if (isset($port_managment['management']) && property_exists($port_managment['management'], 'ipaddress')) {
```

File: src/Inventory/Asset/Printer.php [hunk 2]
```diff
@@ -153,6 +153,31 @@ public function prepare(): array
...
+            foreach ($this->getManagementPorts()['management']->ipaddress as $pa_ip_key => $pa_ip_val) {
```
Comment (trasher) [inline src/Inventory/Asset/Printer.php:160]:
```suggestion
            foreach ($port_managment['management']->ipaddress as $pa_ip_key => $pa_ip_val) {
```

File: src/Inventory/Asset/Printer.php (hunk 1 above)
Comment (stonebuzz) [reply src/Inventory/Asset/Printer.php:159]: ```$port_managment``` is not used to fetch data, it's used to unset data if need, then is set by ```setManagementPorts```

File: src/Inventory/Asset/Printer.php (hunk 2 above)
Comment (stonebuzz) [reply src/Inventory/Asset/Printer.php:160]: ```$port_managment``` is not used to fetch data, it's used to unset data if need, then is set by ```setManagementPorts```

File: tests/functionnal/Glpi/Inventory/Assets/Printer.php [hunk 3]
```diff
@@ -136,7 +136,7 @@ public function testSnmpPrinter()
 
        //get one management port only, since iftype 24 is not importable per default
         $this->array($main->getNetworkPorts())->isIdenticalTo([]);
-        $this->array($mports = $main->getManagementPorts())->hasSize(1)->hasKey('management');
+        $this->array($mports = $main->getManagementPorts())->hasSize();
```
Comment (trasher) [inline tests/functionnal/Glpi/Inventory/Assets/Printer.php:139]: You should probably set an expected size here

**Comment by stonebuzz** [comment]
Need this -> https://github.com/glpi-project/glpi/pull/12181
to continue

File: tests/functionnal/Glpi/Inventory/Assets/Printer.php [hunk 4]
```diff
@@ -135,19 +135,9 @@ public function testSnmpPrinter()
             'last_pages_counter' => 1802,
         ]);
 
-       //get one management port only, since iftype 24 is not importable per default
+       //no managmenet port (network_device->ips same as network_port->ips)
```
Comment (orthagh) [inline tests/functionnal/Glpi/Inventory/Assets/Printer.php:138]:
```suggestion
       //no management port (network_device->ips same as network_port->ips)
```

**Review by orthagh — APPROVED** [review]

File: tests/functionnal/Glpi/Inventory/Assets/Printer.php (hunk 4 above)
Comment (stonebuzz) [reply tests/functionnal/Glpi/Inventory/Assets/Printer.php:138]: fix comment

**Review by trasher — APPROVED** [review]

**Pull request merged**